reporter/iface.go

```diff
@@ -33,12 +33,17 @@ type TraceReporter interface {
 	ReportTraceEvent(trace *libpf.Trace, meta *samples.TraceEventMeta) error
 }
 
+type FileOpener interface {
+	// OpenFile opens the given file from the process namespace.
+	OpenFile(file string) (process.ProcessFile, error)
+}
+
 type ExecutableMetadata struct {
 	// MappingFile is the reference to mapping file data.
 	MappingFile libpf.FrameMappingFile
 
 	// Process is the interface to the process holding the file.
-	Process process.Process
+	Opener FileOpener
```

Comment:
I don't see any benefit in this change since `Opener` is a subset interface of `Process`. And the interface method still returns `process` package specific things. Is there some particular reason why this is done?